src/deploy/osp_deployer/powerflexgw.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2015-2020 Dell Inc. or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import re
+import subprocess
+import sys
+import tempfile
+import time
+import yaml
+from osp_deployer.settings.config import Settings
+from infra_host import InfraHost
+from infra_host import directory_check
+from auto_common import Scp
+from director import Director
+
+logger = logging.getLogger("osp_deployer")
+
+class Powerflexgw(InfraHost):
+
+
+    def __init__(self):
+
+        self.settings = Settings.settings
+        logger.info("Settings.settings: %s", str(Settings.settings))
+        self.ip = self.settings.powerflexgw_vm.public_api_ip
+        self.root_pwd = self.settings.powerflexgw_vm.root_password
+        self.rpm_dir = "/root/rpms"
+        self.certs_dir = "/root/certs"
+        self.lia_cert = "/opt/emc/scaleio/lia/cfg/lia_certificate.pem"
+        self.mdm_cert = "/opt/emc/scaleio/mdm/cfg/mdm_management_certificate.pem"
+        self.keystore = "/opt/emc/scaleio/gateway/webapps/ROOT/WEB-INF/classes/certificates/truststore.jks"
+        self.keystore_pwd = "changeit"
+        self.controller_vip = self.settings.provisioning_vip
+        self.user = self.settings.director_install_account_user
+        self.home_dir = "/home/" + self.user 
+        self.source_overcloudrc = "source " + self.home_dir + "/" + self.settings.overcloud_name + "rc; " 
+
+
+    def upload_rpm(self):
+
+        cmd = "mkdir -p " + self.rpm_dir
+        self.run_as_root(cmd)
+        
+        logger.debug("Uploading powerflex gateway rpm")
+        source_file = "/root/JetPack/src/pilot/powerflex/rpms/" + \
+                      self.settings.powerflex_gateway_rpm
```

Comment:
We should have a setting holding  reference to the path pilot is under already, can we use that instead of hardcoding the full path here

Reply:
Found this: self.foreman_configuration_scripts = self.cloud_repo_dir + "/src"
But nothing for the pilot path.

Reply:
Removing hard coding of the path. Using foreman_configuration_scripts as the base